src/install-peerdeps.js

```diff
@@ -125,32 +114,60 @@ function getPackageJson(
       );
     }
   }
-  return getPackageData({ packageName, registry, auth, proxy }).then(data => {
-    const versions = Object.keys(data.versions);
-    // Get max satisfying semver version
-    let versionToInstall = maxSatisfying(versions, version);
-    // If we didn't find a version, maybe it's a tag
-    if (versionToInstall === null) {
-      const tags = Object.keys(data["dist-tags"]);
-      //  If it's not a valid tag, throw an error
-      if (tags.indexOf(version) === -1) {
-        throw new Error("That version or tag does not exist.");
-      }
-      // If the tag is valid, then find the version corresponding to the tag
-      versionToInstall = data["dist-tags"][version];
-    }
 
-    return data.versions[versionToInstall];
-  });
+  // Remote registry
+  return getPackageData({ packageName, packageManager, version })
+    .then(data => {
+      return Promise.resolve(
+        findPackageVersion({
+          data,
+          version
+        })
+      );
+    })
+    .then(version => {
+      return getPackageData({
+        packageName,
+        packageManager,
+        version
+      });
+    });
 }
 
+/**
+ * Builds the package install string based on the version
+ * @param {Object} options - information needed to build a package install string
+ * @param {string} options.name - name of the package
+ * @param {string} options.version - version string of the package
+ * @returns {string} - the package name and version formatted for an install command
+ */
+const getPackageString = ({ name, version }) => {
+  // check for whitespace
+  if (version.indexOf(" ") >= 0) {
+    // Semver ranges can have a join of comparator sets
+    // e.g. '^3.0.2 || ^4.0.0' or '>=1.2.7 <1.3.0'
+    // Take each version in the range and find the maxSatisfying
+    const rangeSplit = version
+      .split(" ")
+      .map(v => coerce(v))
+      .filter(v => valid(v));
+    const versionToInstall = maxSatisfying(rangeSplit, version);
```

Comment:
can this use `semver` instead of trying to split the version manually?

Reply:
I think that’s out of scope for my PR. I’m. It super familiar with server intricacies and don’t want to break what this original code is doing. I didn’t modify this, just moved it